src/app/container/register-tool/register-tool.service.ts

```diff
@@ -0,0 +1,207 @@
+        this.tool.next(newTool);
+    }
+
+    setToolRegisterError(error: any) {
+        let mapthing = null;
+        if (error) {
+        mapthing = {
+            message: 'The webservice encountered an error trying to create this ' +
+            'container, please ensure that the container attributes are ' +
+            'valid and the same image has not already been registered.',
+            errorDetails: '[HTTP ' + error.status + '] ' + error.statusText + ': ' +
+            error._body
+        };
+    }
+    this.toolRegisterError.next(mapthing);
+    }
+
+    registerTool(newTool: Tool, customDockerRegistryPath) {
+        console.log('Registering tool');
+        this.setTool(newTool);
+        const normalizedToolObj = this.getNormalizedToolObj(newTool, customDockerRegistryPath);
+        this.containerWebService.postRegisterManual(normalizedToolObj).subscribe(response => {
+            // Use types instead
+            (<any>$('#registerContainerModal')).modal('toggle');
+        }, error => this.setToolRegisterError(error)
+        );
+    }
+
+    setCustomDockerRegistryPath(newCustomDockerRegistryPath: string): void {
+        this.customDockerRegistryPath.next(newCustomDockerRegistryPath);
+    }
+
+    setShowCustomDockerRegistryPath(newShowCustomDockerRegistoryPath: boolean): void {
+        this.showCustomDockerRegistryPath.next(newShowCustomDockerRegistoryPath);
+    }
+
+    isInvalidPrivateTool() {
+        return false;
+    }
+    isInvalidCustomRegistry() {
+        return false;
+    }
+
+    getImagePath(imagePath, part) {
+        /** Defines the regex that an image path (namespace/name) must match.
+         Group 1 = namespace, Group 2 = name*/
+        const imagePathRegexp = /^(([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)|_)\/([a-zA-Z0-9]+([-_.][a-zA-Z0-9]+)*)$/i;
+        const matchObj = imagePath.match(imagePathRegexp);
+        let imageName = '';
+        if (matchObj && matchObj.length > 2) {
+            imageName = (part !== 'name') ? matchObj[1] : matchObj[4];
+        }
+        return imageName;
+    };
+
+    getGitUrl(gitPath, scrProvider) {
+        let gitUrl = '';
+        switch (scrProvider) {
+            case 'GitHub':
+                gitUrl = 'https://github.com/';
+                break;
+            case 'GitLab':
+                gitUrl = 'https://gitlab.com/';
+                break;
+            case 'Bitbucket':
+            /* falls through */
+            default:
+                gitUrl = 'https://bitbucket.org/';
+        }
+        gitUrl += gitPath;
+        return gitUrl;
+    };
+
+    createPath(toolObj: Tool, customDockerRegistryPath) {
+        let path = '';
+        if (customDockerRegistryPath !== null) {
+            path += customDockerRegistryPath;
+        } else {
+            path += this.getImageRegistryPath(toolObj.irProvider);
+        }
+        path += '/' + this.getImagePath(toolObj.imagePath, 'namespace') + '/' + this.getImagePath(toolObj.imagePath, 'name');
```

Comment:
How does this deal with alternate tool names when two tools belong to the same repo?